pkg/interface/chat/src/js/components/chat.js

```diff
@@ -194,23 +204,34 @@ export class ChatScreen extends Component {
     let reversedMessages = messages.reverse();
     let chatMessages = reversedMessages.map((msg, i) => {
       // Render sigil if previous message is not by the same sender
-      let gamAut = ['gam', 'aut'];      
+      let gamAut = ['gam', 'aut'];    
+
+      // Local messages don't have a 'gam' prop, so look for the top level if it doesn't exist.
+      let aut = msg.aut ? msg.aut : null;
+      
+      // No gamAut? Return top level author for the same sender check.
       let renderSigil =
-        _.get(reversedMessages[i + 1], gamAut) !== _.get(msg, gamAut);
+        _.get(reversedMessages[i + 1], gamAut) !== _.get(msg, gamAut, aut);
 
       // More padding top if previous message is not by the same sender
       let paddingTop = renderSigil;
       // More padding bot if next message is not by the same sender
       let paddingBot =
-        _.get(reversedMessages[i - 1], gamAut) !== _.get(msg, gamAut);
+        _.get(reversedMessages[i - 1], gamAut) !== _.get(msg, gamAut, aut);
+
+      // Non-local ships don't have pending props.
+      if (!pending) {
```

Comment:
Where is this pending coming from? I don't see it defined earlier in the file.

Reply:
"pending" is only set for local messages, which have it set to "true". So remote messages, if it doesn't exist at all, are given a default value.

Honestly, given that we just remove pending messages without regarding the boolean at all, I could just remove the pending `pending = true` boolean logic entirely from this and the other functions?

Reply:
Tried going without this section -- but this is still necessary or Chat crashes because pending can't be found. Even if I just do a `!!pending` call for the prop.

Reply:
I still don't understand where pending is coming from honestly. It seems like you'd want to check `msg.pending` or something laong those lines??

Reply:
... You're correct. Resolving to check msg.pending.